packages/check-time/extra/check-time.c

```diff
@@ -0,0 +1,41 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/timex.h>
+#include <string.h>
+#include <errno.h>
+
+#define USEC_PER_MSEC 1000L
+#define MAX_EST_ERROR_US 100L * USEC_PER_MSEC /* 100 millisecond */
+
+int main() {
+	struct timex tx;
+	int rc;
+	long long int error;
+
+	if (getenv("IGNORE_CHECK_TIME")) {
+		fprintf(stderr, "check-time ignoring time, passing no matter what\n");
+		return 0;
+	}
+
+	memset(&tx, 0, sizeof(tx));
+	rc = adjtimex(&tx);
+	error = tx.esterror - MAX_EST_ERROR_US;
```

Comment:
Shouldn't access tx.esterror until return code has been checked.

tx.esterror may be positive or negative, so the "are we in sync enough" should probably be
```
/* Make error value absolute, avoiding abs() which is a bit odd as a function (or macro sometimes!) */
long abs_error = tx.esterror > 0 ? tx.esterror : tx.esterror
if (abs_error > MAX_EST_ERROR_US) {
  // print error message
}
```